src/devices/Qmc5883l/Qmc5883l.cs

```diff
@@ -0,0 +1,123 @@
+        /// QMC5883L Heading (DEG)
+        /// </summary>
+        public Angle Heading => VectorExtentsion.GetHeading(ReadDirectionVector());
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Qmc5883l"/> class.
+        /// </summary>
+        public Qmc5883l(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+        }
+
+        /// <summary>
+        /// Sets the sensors mode with additional parameters.
+        /// </summary>
+        /// <param name="mode">Sensor mode</param>
+        /// <param name="interrupt">Controlls the Interrupt PIN</param>
+        /// <param name="rollPointer">When the point roll-over function is enabled, the I2C data pointer automatically rolls between 00H ~ 06H</param>
+        /// <param name="outputRate">Typical Data Output Rate (Hz)</param>
+        /// <param name="oversampling">Bandwidth of an internal digital filter</param>
+        /// <param name="fieldRange">The field range goes hand in hand with the sensitivity of the magnetic sensor</param>
+        public void SetMode(Mode mode = Mode.Continuous,
+            Interrupt interrupt = Interrupt.Disable,
+            RollPointer rollPointer = RollPointer.Disable,
+            OutputRate outputRate = OutputRate.Rate10Hz,
+            Oversampling oversampling = Oversampling.Rate128,
+            FieldRange fieldRange = FieldRange.Gauss2)
+        {
+            Span<byte> interruptCommand = stackalloc byte[]
+            {
+                (byte)Registers.QMC_CONFIG_REG_2_ADDR,
+                (byte)rollPointer
+            };
+            Span<byte> resetCommand = stackalloc byte[]
+            {
+                (byte)Registers.QMC_RESET_REG_ADDR,
+                0x01
+            };
+            byte config = (byte)((byte)mode | (byte)outputRate | (byte)fieldRange | (byte)oversampling);
+            Console.WriteLine(config);
+            Span<byte> setMainRegisteryCommand = stackalloc byte[]
+            {
+                (byte)Registers.QMC_CONFIG_REG_1_ADDR,
+                config
+            };
+            _i2cDevice.Write(interruptCommand);
+            _i2cDevice.Write(resetCommand);
+            _i2cDevice.Write(setMainRegisteryCommand);
+        }
+
+        /// <summary>
+        /// Reads status register to determine if data is avaliable.
+        /// </summary>
+        /// <returns>1 == read is ready. 0 == read is not ready</returns>
+        public bool IsReady()
+        {
+            _i2cDevice.WriteByte((byte)Registers.QMC_STATUS_REG_ADDR);
+            byte status = _i2cDevice.ReadByte();
+            Console.WriteLine(status);
+            return (status & 0x01) == 1;
+        }
+
+        /// <summary>
+        /// Read raw data from QMC5883L
+        /// </summary>
+        /// <returns>Raw Data</returns>
+        private Vector3 ReadDirectionVector()
+        {
+            Span<byte> xRead = stackalloc byte[2];
```

Comment:
This can be called before SetMode() was used. Either call `SetMode()` in the constructor or throw an exception here if the device is not initialized (or not ready)